src/tests/CTABlock.test.tsx

```diff
@@ -0,0 +1,91 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { CTABlock } from '../components/CTABlock'
+import { parseCTABlock, OS_RAISE_IB_EVENT_PREFIX } from '../lib/blocks'
+import { mathifyElement } from '../lib/math'
+import '@testing-library/jest-dom'
+
+jest.mock('../lib/math.ts', () => ({
+  mathifyElement: jest.fn()
+}))
+
+test('CTABlock renders', async () => {
+  render(
+    <CTABlock buttonText="Click me!" contentString={'<p>String</p>'} contentPrompt={'<p>Prompt</p>'}/>
+  )
+
+  screen.getByText('String')
+  screen.getByText('Prompt')
+  screen.getByText('Click me!')
+  screen.getByRole('button')
+})
+
+test('CTABlock fires event', async () => {
+  render(
+      <CTABlock buttonText="Click me!" contentString={'<p>String</p>'} contentPrompt={'<p>Prompt</p>'} fireEvent={'Event'}/>
+  )
+  const eventHandler = jest.fn()
+
+  document.addEventListener('Event', eventHandler)
+  fireEvent.click(screen.getByText('Click me!'))
+  expect(eventHandler).toHaveBeenCalled()
+})
+
+test('CTABlock button and prompt disappear on click', async () => {
+  render(
+      <CTABlock buttonText="Click me!" contentString={'<p>String</p>'} contentPrompt={'<p>Prompt</p>'} fireEvent={'Event'}/>
+  )
+
+  fireEvent.click(screen.getByText('Click me!'))
+  expect(screen.queryByText('Click me!')).toBeNull()
+  screen.getByText('String')
+  expect(screen.queryByText('Prompt')).toBeNull()
+})
+
+test('CTABlock does not render if waitForEvent does not fire', async () => {
+  render(
+      <CTABlock buttonText="Click me!" contentString={'<p>String</p>'} contentPrompt={'<p>Prompt</p>'} waitForEvent={'waitForEvent'}/>
+  )
+
+  expect(screen.queryByText('String')).toBeNull()
+  expect(screen.queryByText('Prompt')).toBeNull()
+  expect(screen.queryByText('Click me!')).toBeNull()
+})
+
+test('CTABlock does render if waitForEvent is fired', async () => {
+  render(
+      <CTABlock buttonText="Click me!" contentString={'<p>String</p>'} contentPrompt={'<p>Prompt</p>'} waitForEvent={'waitForEvent'}/>
+  )
+  const renderEvent = new CustomEvent('waitForEvent')
+
+  fireEvent(document, renderEvent)
+
+  screen.queryByText('String')
+  screen.queryByText('Prompt')
+  screen.queryByText('Click me!')
+})
+
+test('CTABlock from parseCTABlock renders on namespaced event', async () => {
```

Comment:
It might be nice to include a corresponding `CTABlock from parseCTABlock fires namespaced event on click`